### PR TITLE
Added OpInfo for torch.sum

### PR DESCRIPTION
**Pull request by heitorschueroff**

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1046,6 +1046,60 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
...
+    for t in _generate_reduction_inputs(device, dtype, requires_grad):
+        inputs.append(SampleInput(t))
+        for kwargs in _generate_reduction_kwargs(t.ndim):
+            inputs.append(SampleInput(t, kwargs=kwargs))
+
+    return inputs
+
+def sample_inputs_reduction_sum(op_info, device, dtype, requires_grad):
+    inputs = sample_inputs_reduction(op_info, device, dtype, requires_grad)
+    inputs[-1].kwargs['dtype'] = torch.double
```
Comment (heitorschueroff) [inline torch/testing/_internal/common_methods_invocations.py:1100]: Change to cdouble for complex inputs

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1046,6 +1046,60 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
...
+def _generate_reduction_inputs(device, dtype, requires_grad):
+    yield make_tensor((), device, dtype, requires_grad=requires_grad)
+    yield make_tensor((2,), device, dtype, requires_grad=requires_grad)
+    yield make_tensor((2, 2), device, dtype, requires_grad=requires_grad, discontiguous=True)
+    yield make_tensor((3, 2, 1, 2, 2), device, dtype, requires_grad=requires_grad)
+
+# Generates a subset of possible kwargs dim and keepdim for a tensor
+# with ndim dims appropriate for testing. If supports_multiple_dims
+# is True (default) then dim kwarg can be a list of dims.
+def _generate_reduction_kwargs(ndim, supports_multiple_dims=True):
```
Comment (heitorschueroff) [inline torch/testing/_internal/common_methods_invocations.py:1059]: Replace with manual cases
Include inner and outer most dimensions

**Comment by heitorschueroff** [comment]
@mruberry  I updated with your suggestions.

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3228,6 +3274,10 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_out=False,
            sample_inputs_func=sample_inputs_max_min_reduction_no_dim,),
+    OpInfo('sum',
+           dtypes=all_types_and_complex_and(torch.float16, torch.bfloat16, torch.bool),
+           supports_out=False,
+           sample_inputs_func=sample_inputs_reduction),
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:3280]: probably add a TODO for adding tests for kwarg

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -3228,6 +3274,10 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_out=False,
            sample_inputs_func=sample_inputs_max_min_reduction_no_dim,),
+    OpInfo('sum',
+           dtypes=all_types_and_complex_and(torch.float16, torch.bfloat16, torch.bool),
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:3278]: should we also compare this against the numpy function: https://numpy.org/doc/stable/reference/generated/numpy.sum.html ?

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3278]: The base OpInfo class doesn't support references, unfortunately. I'm sure ReductionInfos will, though ;)

**Review by mruberry — APPROVED** [review]
Nice improvements! Let's get this landed ahead of tomorrow.

**Pull request closed without merging**